docs/topics/security/authentication/README.md

```diff
@@ -0,0 +1,117 @@
+# Authentication
+
+The driver can be setup to use either the built-in plain text authentication
+or to use a custom authentication implementation.
+
+## Plain text
+
+Credentials are provided using the [`cass_cluster_set_credentials()`] function.
+
+```c
+cass_cluster_set_credentials(cluster, "cassandra", "cassandra");
+```
+
+**Important**: The credentials are sent in plain text to the server. For this
+reason, it is highly recommended that this be used in conjunction with
+client-to-node encryption (SSL), or in a trusted network environment.
+
+## Custom
+
+A custom authentication implementation can be set using
+[`cass_cluster_set_auth_callbacks()`]. This is useful for integrating with more
+complex authentication systems such as Kerberos.
+
+```c
+size_t on_auth_initial(CassAuth* auth,
+                       void* data,
+                       char* response,
+                       size_t response_size) {
+  /*
+   * This callback is used to initiate a request to begin an authentication
+   * exchange. Required resources can be acquired and initialized here.
+   *
+   * Resources required for this specific exchange can be stored in the
+   * `auth->data` field and will be available in the subsequent challenge
+   * and success phases of the exchange. The cleanup callback should be used to
+   * free these resources. This data will only ever be accessed by a single
+   * thread.
```

Comment:
Hmmm. The  the `data` argument is different from the `auth->data`. I need to figure out how to make that not confusing....